### ISSUE TITLE
iOS privacy manifest

### ISSUE THREAD

**Issue by gabrielguilhoto** [itinance/react-native-fs#1232]
Starting May 1, Apple will require iOS submissions to the App Store to have privacy manifests describing the use of required reason APIs: https://developer.apple.com/news/?id=3d8a9yyh.

I used [a script](https://github.com/Wooder/ios_17_required_reason_api_scanner) to find required reason APIs in our project and it pointed to 4 uses by `react-native-fs` in the `RNFSManager.m` file, at least in version 2.19.0:

* `NSFileCreationDate`
* `NSFileModificationDate`
* `NSFileSystemFreeSize`
* `NSFileSystemSize`

Could you please add the required privacy manifest?

**Comment by gkasireddy202** [comment]
How to download sh required_reason_api_scanner.sh YourAppDirectory and run in your react-native project?

**Comment by gabrielguilhoto** [comment]
> How to download sh required_reason_api_scanner.sh YourAppDirectory and run in your react-native project?

You can [clone](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository) the https://github.com/Wooder/ios_17_required_reason_api_scanner repository to be able to run it.

**Comment by gkasireddy202** [comment]
@gabrielguilhoto - Thanks for your reply.

**Comment by gkasireddy202** [comment]
@gabrielguilhoto -  After I cloned the git on my desktop.Getting error.
sh required_reason_api_scanner.sh Desktop 

sh: required_reason_api_scanner.sh: No such file or directory

**Comment by WilliamWelsh** [comment]
> @gabrielguilhoto - After I cloned the git on my desktop.Getting error. sh required_reason_api_scanner.sh Desktop
> 
> sh: required_reason_api_scanner.sh: No such file or directory

If you cloned it on your desktop then there is no folder called Desktop... you would just put `.`...

**Comment by gkasireddy202** [comment]
@WilliamWelsh  - Thanks it is working. Is it mandatory to update every library for iOS privacy manifest?

**Comment by gkasireddy202** [comment]
I copied the required_reason_api_text_scanner.sh in my project and ran the sh required_reason_api_text_scanner.sh .
Showing only the potentially required reason API usage 'UserDefaults' in './projectname/AppDelegate.m'
Line numbers: 11 21 but my project has many libraries like react-native-fs,react-native-calendar,react-native-google-maps-directions...

**Comment by birdofpreyru** [comment]
Yeah... should I mention it once again, [#1197 &mdash; The future of **react-native-fs**](https://github.com/itinance/react-native-fs/issues/1197)?
You probably should migrate to my fork of the lib, and file new issues, do any contributions in there.

**Comment by gkasireddy202** [comment]
I am using   react-native-fs:2.20.0.
Can I update this library for the iOS privacy manifest file?

**Comment by stami** [comment]
I have no use for disk space APIs so I removed them with [patch-package](https://github.com/ds300/patch-package):
```diff
diff --git a/node_modules/react-native-fs/RNFSManager.m b/node_modules/react-native-fs/RNFSManager.m
index 5ddd941..80fb3e9 100755
--- a/node_modules/react-native-fs/RNFSManager.m
+++ b/node_modules/react-native-fs/RNFSManager.m
@@ -726,24 +726,11 @@ + (BOOL)requiresMainQueueSetup
 {
   unsigned long long totalSpace = 0;
   unsigned long long totalFreeSpace = 0;
-
-  __autoreleasing NSError *error = nil;
-  NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
-  NSDictionary *dictionary = [[NSFileManager defaultManager] attributesOfFileSystemForPath:[paths lastObject] error:&error];
-
-  if (dictionary) {
-    NSNumber *fileSystemSizeInBytes = [dictionary objectForKey: NSFileSystemSize];
-    NSNumber *freeFileSystemSizeInBytes = [dictionary objectForKey:NSFileSystemFreeSize];
-    totalSpace = [fileSystemSizeInBytes unsignedLongLongValue];
-    totalFreeSpace = [freeFileSystemSizeInBytes unsignedLongLongValue];
-
-    resolve(@{
-      @"totalSpace": [NSNumber numberWithUnsignedLongLong:totalSpace],
-      @"freeSpace": [NSNumber numberWithUnsignedLongLong:totalFreeSpace]
-    });
-  } else {
-    [self reject:reject withError:error];
-  }
+  // Remove disk space APIs to not to trigger NSPrivacyAccessedAPICategoryDiskSpace for privacy manifest
+  resolve(@{
+    @"totalSpace": [NSNumber numberWithUnsignedLongLong:totalSpace],
+    @"freeSpace": [NSNumber numberWithUnsignedLongLong:totalFreeSpace]
+  });
 }
 
 

```

**Comment by MianSaad705** [comment]
@gabrielguilhoto @gkasireddy202  Have You found any workaround for this or do we have to manually add the reason in PrivacyInfo.xcprivacy under this API Type NSPrivacyAccessedAPICategoryFileTimestamp.

**Comment by gkasireddy202** [comment]
> Have You found any workaround for this or do we have to manually add the reason in PrivacyInfo.xcprivacy under this API Type NSPrivacyAccessedAPICategoryFileTimestamp.

I added this API reason in PrivacyInfo.xcprivacy file in my app.

**Comment by MianSaad705** [comment]
@gkasireddy202  Under this API NSPrivacyAccessedAPICategoryFileTimestamp?
Secondly after adding apple approved your app without any warnings?
Kindly mention which reason you add 

- DDA9.1
- C617.1
- 3B52.1
- 0A2A.1

https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api

**Comment by gkasireddy202** [comment]
@MianSaad705  - I added the below API type and reason in my app privacy file.I did not get any warning emails from Apple and also app is approved.
<dict>
			<key>NSPrivacyAccessedAPIType</key>
			<string>NSPrivacyAccessedAPICategoryFileTimestamp</string>
			<key>NSPrivacyAccessedAPITypeReasons</key>
			<array>
				<string>C617.1</string>
			</array>
		</dict>

**Comment by MianSaad705** [comment]
@gkasireddy202  Thanks.
